spec/models/post_revision_spec.rb

```diff
@@ -84,4 +84,21 @@
     expect(post_revision.valid?).to eq(true)
     expect(post_revision.user_id).to eq(SiteSetting.get(:staff_alias_user_id))
   end
+
+  it "switches revision user to staff alias user when changing user_id of post to staff alias user" do
+    post = Fabricate(:post, user: user, post_type: Post.types[:regular])
+
+    post_revision =
+      Fabricate.build(
+        :post_revision,
+        post: post,
+        user: DiscourseStaffAlias.alias_user,
```

Comment:
Should this be set to another user so that we can assert that the user indeed changed?

Reply:
I believe this should be set to a user that satisfy `DiscourseStaffAlias.user_allowed?(self.user)`

Reply:
> I believe this should be set to a user that satisfy DiscourseStaffAlias.user_allowed?(self.user)

I mentioned this above and I felt like the restriction is unnecessary. A moderator or users who are explicitly allowed to edit posts feels like they should be allowed to do this.

> set to another user so that we can assert that the user indeed changed

✅